File: syJoh/BOJ/숫자의 개수.py
a = int(input())
b = int(input())
c = int(input())
arr = list(str(a*b*c))

for i in range(10):
    print(arr.count(str(i)))

"""
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int a = Integer.parseInt(br.readLine());
        int b = Integer.parseInt(br.readLine());
        int c = Integer.parseInt(br.readLine());
        int[] arr = new int[10];
        String[] tmp = String.valueOf(a*b*c).split("");
        for(int i=0; i<tmp.length; i++){
            arr[Integer.parseInt(tmp[i])]++;
        }
        for(int num: arr){
            System.out.println(num);
        }
        br.close();
    }
}
"""

File: ARaC/BOJ_BD/숫자의개수.java

import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int a = scanner.nextInt();
        int b = scanner.nextInt();
        int c = scanner.nextInt();
        solution(a,b,c);

    }
    public static void solution(int a, int b, int c){
        String str = String.valueOf(a*b*c);
        int[] counts = {0,0,0,0,0,0,0,0,0,0};

        int num = 0;
        for(int i =0; i< str.length(); i++){

            num = Integer.parseInt(String.valueOf(str.charAt(i)));
            counts[num] += 1;
        }

        for(int n : counts){
            System.out.println(n);
        }
    }
}
